src/main/java/com/amazonaws/encryptionsdk/internal/Utils.java

```diff
@@ -268,7 +268,7 @@ public static ByteBuffer limit(final ByteBuffer buff, final int newLimit) {
      * @return decoded data as a byte array
      */
     public static byte[] decodeBase64String(final String encoded) {
-        return Base64.decode(encoded);
+        return encoded.isEmpty() ? new byte[0] : Base64.decode(encoded);
```

Comment:
We should already have a constant defined for a zero-byte array. Return that instead. Otherwise you create needless memory pressure.

Reply:
I didn't see a constant defined in Constants.java, so I just used the one in Apache Commons Lang ArrayUtils. If you prefer I can add the constant to Constants.java